apps/web/hooks/useFolders.ts

```diff
@@ -0,0 +1,12 @@
+import type { OutlookFolder } from "@/utils/outlook/folders";
+import useSWR from "swr";
+
+export function useFolders({ emailAccountId }: { emailAccountId: string }) {
```

Comment:
You don't need to pass emailAccountId.

See useRules. It works because of this in SWRProvider

```ts
// https://swr.vercel.app/docs/error-handling#status-code-and-error-object
const fetcher = async (
  url: string,
  init?: RequestInit | undefined,
  emailAccountId?: string | null,
) => {
  const headers = new Headers(init?.headers);

  if (emailAccountId) {
    headers.set(EMAIL_ACCOUNT_HEADER, emailAccountId);
  }
```

Reply:
Actually I don't think the code is using your param at all. The backend doesn't process it.

I would switch this file to be like the other swr hook files. Super simple:

```ts
import useSWR from "swr";
import type { RulesResponse } from "@/app/api/user/rules/route";

export function useRules() {
  return useSWR<RulesResponse, { error: string }>("/api/user/rules");
}
```